lib/active_merchant/billing/gateways/mundipagg.rb

```diff
@@ -327,10 +355,10 @@ def gateway_response_errors(response)
         error_string
       end
 
-      def authorization_from(response, action)
-        return "#{response['customer']['id']}|#{response['id']}" if action == 'store'
+      def authorization_from(charge, action)
+        return "#{charge['customer']['id']}|#{charge['id']}" if action == 'store'
 
-        response['id']
+        charge['id']
       end
```

Comment:
Like you mentioned, you'll encounter downstream runtime errors if `charge_object` returns `nil`. You can head that off with a simple `return unless charge`.